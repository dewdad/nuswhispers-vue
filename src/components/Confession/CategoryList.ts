import Vue from 'vue';
import Component from 'vue-class-component';
import { Category } from '../../store/state';

@Component({
  props: {
    ids: Array,
  },
})
export default class CategoryList extends Vue {
  public ids: string[];

  get categories(): Category[] {
    return this.$store.getters.categoriesByIds(this.ids);
  }
}
