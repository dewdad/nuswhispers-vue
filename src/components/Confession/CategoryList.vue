<template>
  <div v-if="categories.length">
    <template v-for="(category, idx) in categories">
      <router-link :to="'/category/' + category.confessionCategoryId">{{ category.confessionCategory }}</router-link><span v-if="idx + 1 < categories.length">, </span>
    </template>
  </div>
  <div v-else>
    No Category
  </div>
</template>

<script lang="ts" src="./CategoryList.ts" />
