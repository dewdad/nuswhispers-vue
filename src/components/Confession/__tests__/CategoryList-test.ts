import Vue from 'vue';
import { createRenderer } from 'vue-server-renderer';
import Vuex from 'vuex';
import createStore from '../../../store';
import CategoryList from '../CategoryList.vue';

Vue.use(Vuex);

describe('CategoryList', () => {
  const renderer = createRenderer();
  const store = createStore();
  store.state.categories = {
    '1': { confessionCategory: 'Foo', confessionCategoryId: '1' }, // tslint:disable-line
    '2': { confessionCategory: 'Bar', confessionCategoryId: '2' }, // tslint:disable-line
  };

  it('renders correctly with categories', () => {
    const Ctor = Vue.extend(CategoryList);
    const vm = new Ctor({
      propsData: {
        ids: ['1', '2'],
      },
      store,
    });

    renderer.renderToString(vm, (err: Error | undefined, html: string) => {
      expect(html).toMatchSnapshot();
    });
  });

  it('renders correctly with one category', () => {
    const Ctor = Vue.extend(CategoryList);
    const vm = new Ctor({
      propsData: {
        ids: ['1'],
      },
      store,
    });

    renderer.renderToString(vm, (err: Error | undefined, html: string) => {
      expect(html).toMatchSnapshot();
    });
  });

  it('renders correctly with no categories', () => {
    const Ctor = Vue.extend(CategoryList);
    const vm = new Ctor({
      propsData: { ids: [] },
      store,
    });

    renderer.renderToString(vm, (err: Error | undefined, html: string) => {
      expect(html).toMatchSnapshot();
    });
  });
});
