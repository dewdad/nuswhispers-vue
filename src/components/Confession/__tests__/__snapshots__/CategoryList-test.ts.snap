// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList renders correctly with categories 1`] = `
<div data-server-rendered="true">
    <router-link to="/category/1">Foo</router-link><span>, </span>
    <router-link to="/category/2">Bar</router-link>
    <!---->
</div>
`;

exports[`CategoryList renders correctly with no categories 1`] = `
<div data-server-rendered="true">
    No Category
</div>
`;

exports[`CategoryList renders correctly with one category 1`] = `
<div data-server-rendered="true">
    <router-link to="/category/1">Foo</router-link>
    <!---->
</div>
`;
